examples/settings/settings_main.c

```diff
@@ -0,0 +1,276 @@
+
+/****************************************************************************
+ * Private Functions
+ ****************************************************************************/
+
+/****************************************************************************
+ * settings_main
+ ****************************************************************************/
+
+int main(int argc, FAR char *argv[])
+{
+  int                  ret;
+  int                  fd;
+  enum settings_type_e stype;
+  char                 path[50];
+  char                 *teststr = "";
+  int                  testval = 0x5aa5;
+  char                 readstr[CONFIG_SETTINGS_VALUE_SIZE];
+  enum storage_type_e  storage_type = STORAGE_BINARY;
+  bool                 flag_present = false;
+  struct stat          sbuf;
+
+  ASSERT(0);
+  if ((argc < 1) || *argv[1] == 0 || *(argv[1] + 1) == 0)
+    {
+      goto print_help;
+    }
+
+  if (*argv[1] == '-')
+    {
+      flag_present = true;
+      if (*(argv[1] + 1) == 'b')
+        {
+          storage_type = STORAGE_BINARY;
+        }
+      else if (*(argv[1] + 1) == 't')
+        {
+          storage_type = STORAGE_TEXT;
+        }
+      else
+        {
+          goto print_help;
+        }
+    }
+
+  if (flag_present && argc < 2)
+    {
+      goto print_help;
+    }
+
+#ifdef CONFIG_EXAMPLES_SETTINGS_USE_TMPFS
+#  ifndef CONFIG_FS_TMPFS
+#    error TMPFS must be enabled
+#  endif
+#  ifndef CONFIG_LIBC_TMPDIR
+#    error LIBC_TMPDIR must be defined
+#  endif
+  if (stat(CONFIG_LIBC_TMPDIR, &sbuf))
+    {
+      ret = nx_mount(NULL, CONFIG_LIBC_TMPDIR, "tmpfs", 0, NULL);
+      if (ret < 0)
+        {
+          printf("ERROR: Failed to mount tmpfs at %s: %d\n",
+                  CONFIG_LIBC_TMPDIR, ret);
+          goto end;
+        }
+    }
+
+  strcpy(path, CONFIG_LIBC_TMPDIR);
+  strcat(path, "/settings");
+#else
+  strcpy(path, CONFIG_EXAMPLES_SETTINGS_EXISTING_STORAGE);
+  if (path == NULL)
+    {
+      printf("Settings filepath is empty!");
+      goto end;
+    }
+
+  strcat(path, "/settings");
+#endif
+
+  printf("Example of settings usage: %s. Path: %s\n",
+         ((storage_type == STORAGE_TEXT) ? "text" : "binary"), path);
+  printf("-------------------------------------------------------\n");
+
+  ret = settings_init();
+  if (ret < 0)
+    {
+      printf("settings init failed: %d", ret);
+      goto end;
+    }
+
+  ret = settings_setstorage(path, storage_type);
+  if (ret == -ENOENT)
+    {
+      printf("INFO: no existing binary settings file found. Creating it.\n");
+      fd = open(path, O_CREAT);
+      if (fd < 0)
+        {
+          printf("failed to create settings file\n");
+          goto end;
+        }
+
+      close(fd);
+    }
+  else if (ret < 0)
+    {
+      printf("settings setstorage failed: %d\n", ret);
+      goto end;
+    }
+  else
+    {
+      printf("INFO: a settings file was found and loaded\n");
+      ret = settings_get("v1", SETTING_STRING, readstr,
```

Comment:
This call should always fail.

You try to access a setting without creating it first.
It **must** be created, even if a value already exists in the file.

Reply:
But it has been loaded from a file that has been found. Why do you have to create it (again)?